src/overlays/gamestates/ovl_title/z_title.c

```diff
@@ -1,22 +1,183 @@
 /*
  * File: z_title.c
  * Overlay: ovl_title
- * Description:
+ * Description: Nintendo 64 logo shown on startup
  */
 
 #include "z_title.h"
+#include "overlays/gamestates/ovl_opening/z_opening.h"
+#include "static/nintendo_rogo_static/nintendo_rogo_static.h"
 
-extern UNK_TYPE D_01002720;
+void Title_UpdateCounters(TitleContext *this) {
+    if ((this->coverAlpha == 0) && (this->visibleDuration != 0)) {
+        this->timer--;
+        this->visibleDuration--;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_title/Title_UpdateCounters.s")
+        if (this->timer == 0) {
+            this->timer = 400;
+        }
+    } else {
+        this->coverAlpha += this->addAlpha;
+        
+        if (this->coverAlpha <= 0) {
+            this->coverAlpha = 0;
+            this->addAlpha = 12;
+        } else if (this->coverAlpha >= 255) {
+            this->coverAlpha = 255;
+            this->exit = true;
+        }
+    }
+    
+    this->uls = this->ult & 0x7F;
+    this->ult++;
```

Comment:
```suggestion
    this->uls = this->ult & 0x7F;
    this->ult++;
```
What is `uls` and `ult`?

Reply:
You can see them used in the Draw loop. I think they're coordinates on the texture; I assume it's to do with the strange way the shine on the logo is drawn... @Thar0 knows more about this graphics stuff than me.